File: examples/src/WeatherIndicator.java
public class WeatherIndicator {
    void rateWeather(int celsius) {
        String status;
        Color color;
        if (celsius < 5) {
            status = "Saukalt!";
            color = Color.BLUE;
        } else if (celsius >= 5 && celsius <= 20) {
            status = "Geht so!";
            color = Color.ORANGE;
        } else {
            status = "Urlaub!";
            color = Color.RED;
        }
    }

    enum Color {BLUE, ORANGE, RED}
}

File: examples/src/weatherindicator.kt
class WeatherIndicatorKt {
    fun rateWeather(celsius: Int) {
        val (status, color) =
            when {
                celsius < 5 -> "Saukalt!" to Color.BLUE
                celsius in 5..20 ->  "Geht so!" to Color.ORANGE
                else -> "Urlaub!" to Color.RED
            }
    }

    enum class Color { BLUE, ORANGE, RED }
}

